app/controllers/concerns/admin/list_paginated_users.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Admin::ListPaginatedUsers
+  extend ActiveSupport::Concern
+  include Pagy::Backend
+
+  RECORDS_PER_PAGE = 5
+
+  private
+    def list_paginated_users(users:, template:, legacy_template:)
+      pagination, users = pagy_countless(
+        users,
+        limit: params[:per_page] || RECORDS_PER_PAGE,
+        page: params[:page],
+        countless_minimal: true
+      )
+
+      respond_to do |format|
+        format.html do
+          render(
+            inertia: template,
+            props: {
+              users: InertiaRails.merge do
```

Comment:
So much black magic in Inertia, I wouldn't have guessed that `InertiaRails.merge` and `WhenVisible` make a pair 😅 Concise and convenient when you get to know it I guess

Reply:
They don't. It's just InertiaRails' way to [merge props](https://inertia-rails.dev/guide/merging-props) instead of overwriting them. `WhenVisible` on the other hand is nothing more than a simple IntersectionObserver that triggers a visit upon intersecting, We could use a form or a link with a partial reload and achieve the same feature, only requiring the user to click instead of scrolling 🙂